src/garage/torch/modules/mlp_module.py

```diff
@@ -56,39 +54,87 @@ def __init__(self,
                  layer_normalization=False):
         super().__init__()
 
-        self._input_dim = input_dim
-        self._output_dim = output_dim
-        self._hidden_nonlinearity = hidden_nonlinearity
-        self._output_nonlinearity = output_nonlinearity
-        self._layer_normalization = layer_normalization
         self._layers = nn.ModuleList()
 
         prev_size = input_dim
-
         for size in hidden_sizes:
-            layer = nn.Linear(prev_size, size)
-            hidden_w_init(layer.weight)
-            hidden_b_init(layer.bias)
-            self._layers.append(layer)
+            hidden_layers = nn.Sequential()
+            if layer_normalization:
+                hidden_layers.add_module('layer_normalization',
+                                         nn.LayerNorm(prev_size))
+            linear_layer = nn.Linear(prev_size, size)
+            hidden_w_init(linear_layer.weight)
+            hidden_b_init(linear_layer.bias)
+            hidden_layers.add_module('linear', linear_layer)
+
+            if hidden_nonlinearity:
+                hidden_layers.add_module('non_linearity',
+                                         _NonLinearity(hidden_nonlinearity))
+
+            self._layers.append(hidden_layers)
             prev_size = size
 
-        layer = nn.Linear(prev_size, output_dim)
-        output_w_init(layer.weight)
-        output_b_init(layer.bias)
-        self._layers.append(layer)
+        output_layer = nn.Sequential()
+        linear_layer = nn.Linear(prev_size, output_dim)
+        output_w_init(linear_layer.weight)
+        output_b_init(linear_layer.bias)
+        output_layer.add_module('linear', linear_layer)
 
-    def forward(self, input_val):
-        """Forward method."""
-        x = input_val
-        for layer in self._layers[:-1]:
-            x = layer(x)
-            if self._hidden_nonlinearity is not None:
-                x = self._hidden_nonlinearity(x)
-            if self._layer_normalization:
-                x = nn.LayerNorm(x.shape[1])(x)
+        if output_nonlinearity:
+            output_layer.add_module('non_linearity',
+                                    _NonLinearity(output_nonlinearity))
+        self._layers.append(output_layer)
+
+    # pylint: disable=arguments-differ
+    def forward(self, input_value):
```

Comment:
please document shape constraints (if any) in the parameter documentation. See the torch documentation for examples.